lib/util/source-code.js

```diff
@@ -88,68 +88,66 @@ function looksLikeExport(astNode) {
  * @param {ASTNode} ast - The Program node of the AST representing the code. This AST should be created from the text that BOM was stripped.
  * @constructor
  */
-function SourceCode(text, ast) {
-    validate(ast);
+class SourceCode {
```

Comment:
wow thats new. People do that too `(context.getSourceCode().getText)()`... lol
I will revert the changes which is on the API and leave the others.